### PR TITLE
[protractor-console-plugin] add type definition

### PR DESCRIPTION
**Pull request by akwiatek**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## protractor-console-plugin ([<kbd>unpkg</kbd>](https://unpkg.com/browse/protractor-console-plugin@latest/))
was missing the following properties:
1. teardown
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 00ff8309badcc957ce3d60f6b268fa2aadd1c69b
</p>

File: types/protractor-console-plugin/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Type definitions for protractor-console-plugin 0.1
+// Project: https://github.com/angular/protractor-console-plugin
+// Definitions by: Adam Kwiatek <https://github.com/akwiatek>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface ProtractorConsolePluginConfig {
```
Comment (akwiatek) [inline types/protractor-console-plugin/index.d.ts:6]: Based on https://github.com/angular/protractor-console-plugin/blob/07672cfb635501a6d48dd977eddb2388f34af827/README.md?plain=1#L10

Comment (peterblazejewicz) [reply types/protractor-console-plugin/index.d.ts:6]: IMO this looks like CommonJS module, with exported public interface of the class-like module (old function-constructor syntax being used). There are module level methods (a.k.a. static) exported automatically through instance as well, defined directly on 'ConsolePlugin)
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
https://github.com/angular/protractor-console-plugin/blob/07672cfb635501a6d48dd977eddb2388f34af827/index.js#L117

**Comment by peterblazejewicz** [comment]
@akwiatek I think I misunderstood the intent of the PR. It's not about the declaration for a module itself, but what you want is to expose a config for this plugin features, e.g. to expose the configuraton types:
```ts
exports.config = {
  plugins: [{
    package: 'protractor-console-plugin',
    failOnWarning: {Boolean}                (Default - false),
    failOnError: {Boolean}                  (Default - true),
    logWarnings: {Boolean}                  (Default - true),
    exclude: {Array of strings and regex}   (Default - [])
  }]
};
```
I guess I missed the intent. The module exports just an instance

**Comment by peterblazejewicz** [comment]
just export the scoped (over the namespace) config type `Config`. Sorry for the fuss.

**Comment by akwiatek** [comment]
@peterblazejewicz 

Following 
https://github.com/angular/protractor-console-plugin/blob/master/README.md#console-plugin-chrome-only
https://github.com/Microsoft/dtslint/blob/master/docs/export-just-namespace.md

I've exported only `Config` from that namespace.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## protractor-console-plugin ([<kbd>unpkg</kbd>](https://unpkg.com/browse/protractor-console-plugin@latest/))
was missing the following properties:
1. teardown
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against acdfca8893a939054e11b3d4f0ba623edd3b127d
</p>

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@akwiatek thanks!

**Comment by akwiatek** [comment]
Ready to merge

**Pull request merged**